docs/proposals/scheduler/README.md

```diff
@@ -0,0 +1,135 @@
+he can handle the needed workload. The controller can also determine in a first time, if a workload can be managed in the cluster, but
+the scheduler has the **last word** on it. 
+
+This service **must** be able at anytime, to request a delete or an update of a workload on a specific
+node. These requests will be sent based on event the scheduler receive. The events are coming either from the controller or from a node.
+
+The `scheduler` is a central point to the whole cluster communication. The service is able to filter
+data coming from nodes and lift them up to the controller. By being a central point of communication
+the `scheduler` can be considered as a [SPOF](https://en.wikipedia.org/wiki/Single_point_of_failure), 
+considering that, this service must be able to recover from a crash **quickly** and **dynamically**.
+
+**Features summary**:
+
+* Allocate a workload to a node
+* Request workload update 
+* Request workload deletion
+* Handle statistics & hardware informations from nodes
+* Lift up informations from nodes to controller 
+* Handle various events from nodes 
+
+**Glossary:**
+* Node: A server of the cluster
+* Node server: The master node which contain cluster essentials components
+* Node worker: A slave node wich receive instruction from master node
+* Worker: entity managing workloads and managed by the scheduler & controller
+* Workload: a unit of work needed to be deployed inside a worker
+* Cluster: the whole architecture containing every single component of this project. 
+* Server / Master: a server containing every administrative services (scheduler & controller)
+
+---
+
+## Architecture overview
+
+![Architecture overview](https://cdn.discordapp.com/attachments/828205813336244254/842476988290695187/a051f160-4d85-4c5c-a49b-0354ca9272d2.png)
+
+## Communication in the cluster 
+
+There are mainly two solutions. Either having direct connections between components (HTTP, gRPC...) of the cluster or having
+message queues where information is centralized.
+
+The second solution can be a great improvement for scalability and performance as everything would be completely asynchronous. 
+However, we need synchronous communication between internal services so make sure every information is properly received & understood.
+That's why the first solution direct communication between major components is a good choice. The drawback of this solution is we have 
+to write APIs for each component. 
+
+On top of that, we will be using [gRPC](https://grpc.io/) for communication between components. It will be handy to use 
+as API definitions are defined through [protoBuf](https://developers.google.com/protocol-buffers). 
+
+There is an interrogation around the controller & scheduler communication. They may be in the same 
+physical machine and not isolated one from another, so do we still need to use gRPC here ? Can't we 
+use any other solution of communication, as we are on the same physical machine ?
```

Comment:
We tried to think about the case where these two won't be on the same machine, but we were not sure if we would have to use another way for local information sharing between components. So it will be `gRPC` everywhere :D

Reply:
> So it will be gRPC everywhere :D

Yes :-)